woocommerce/google-listings-and-ads
# Add Endpoint- Create Asset Group

### Changes proposed in this Pull Request:



Closes part of https://github.com/woocommerce/google-listings-and-ads/issues/1780

This PR adds a new endpoint to create asset groups for a specific ad campaign. By default, all campaigns are created with an asset group but the user can remove the asset group later on, so in case the user removes all asset groups we will be able to create a new one if needed. 

![XD8vmD.png](https://user-images.githubusercontent.com/2488994/212314911-68a7b90d-6c67-4930-a292-3463a878295b.png)

Also, this PR has updated the endpoint route for getting assets from asset groups, the route has been changed from `ads/campaigns/CAMPAIGN_ID/asset-groups` to `ads/campaigns/asset-groups?campaign_id=CAMPAIGN_ID`, I made the change because I think the endpoint name will be more aligned with the other controllers that we have. This endpoint is not yet used for the front-end.


### Detailed test instructions:


1. As the branch `feature/pmax-assets` is using some new imports from the GoogleAds Library, you may need to remove your vendor folder `rm -Rf vendor/*` and install again the dependencies using `composer install`.
2. Make a request to `POST wc/gla/ads/campaigns/asset-groups` with the following body:
```
{
	"campaign_id" : YOUR_CAMPAIGN_ID
}
```
3. The response should look likes this:
```
{
    "status": "success",
    "message": "Successfully created asset group.",
    "id": 6460020972
}
```
4. Check in your Ads account if the asset group has been created or make a request to `GET ads/campaigns/asset-groups?campaign_id=YOUR_CAMPAIGN_ID` (the endpoint only returns one asset group)





### Changelog entry
